examples/map/difference_images.py

```diff
@@ -0,0 +1,63 @@
+"""
+===========================
+Plotting a difference image
+===========================
+
+How to compute and plot a difference image.
+The example uses `sunpy.map.MapSequence` to compute a difference image and then plot it.
+This basic method works for base difference or running difference. Just change whether
+you're subtracting the previous image or the first image in a sequence.
+"""
+###########################################################################
+
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+import astropy.units as u
+from sunpy.net import Fido, attrs as a
+import sunpy.map
+
+###########################################################################
+# First we'll download a couple images and store them in a
+# `sunpy.map.MapSequence`. This could be from any channel of any imager.
+# Here, we use SDO/AIA 304 Å.
+
+instrument = a.Instrument('AIA')
+wave = a.Wavelength(30 * u.nm, 31 * u.nm)
+result = Fido.search(a.Time('2015-06-18T00:00:00', '2015-06-18T00:00:10') |
+                     a.Time('2015-06-18T01:03:30', '2015-06-18T01:03:35'), instrument, wave)
+downloaded_files = Fido.fetch(result)
+maps = sunpy.map.Map(downloaded_files, sequence=True)
+
+###########################################################################
+# Now we'll do a standard plot of the second image just to see it.
+
+plt.figure()
+ax = plt.subplot(projection=maps[1])
+maps[1].plot()
+
+###########################################################################
+# And now we can do take the actual difference.
+
+diff = maps[1].data - maps[0].data
+
+###########################################################################
+# But we have to decide what to do with the metadata. For example, what
+# time does this difference image correspond to? The time of the first or
+# second image? The mean time? You'll have to decide what makes most sense
+# for your application. Here we'll just use the metadata from the second
```

Comment:
Could we set some property in the meta to be custom here, just so that there is something for someone who want's to edit it to copy?

Reply:
I'm not opposed. But you have a much better idea than me what property would be best.

Reply:
I think the only one which would show up on the plot would be `DATE-OBS`.

Reply:
Tried this but it crashes unless `date-obs` is a properly formatted timestamp. I tried this useful thing: 

```
meta = maps[1].meta
timediff = Time(maps[1].meta['date-obs']) - Time(maps[0].meta['date-obs'])
meta['date-obs'] = '{} Minute Difference Image'.format(int(round(timediff.sec / 60.)))
diff_map = sunpy.map.Map(diff, meta)
```

but that causes:
```
ValueError: Input values did not match any of the formats where the format keyword is optional ['datetime', 'iso', 'isot', 'yday', 'datetime64', 'fits', 'byear_str', 'jyear_str']
```

Ditto if I just set `meta['date-obs'] = 'Turtle'`. The error occurs at the line:
```
ax_diff = plt.subplot(projection=diff_map)
```

Reply:
Yeah, 'date-obs` is going to have to follow a specific format. I think setting the `date-obs` as the second time (i.e. in diff time_2 - time_1).

Reply:
Setting it to the second time is what it originally/presently does

Reply:
ok, lets just leave it. :smile: